src/lib/components/utils/acm-tagfield.svelte

```diff
@@ -0,0 +1,128 @@
+<script lang="ts">
+  import { createEventDispatcher } from 'svelte';
+
+  export let tags: string[] = [];
+  export let selectedTags: string[] = [];
+  export let urlSearchParamKey = '';
+
+  const dispatch = createEventDispatcher();
+  let hasSelectedTags = selectedTags.length > 0;
+
+  function selectTag(event: MouseEvent) {
+    event.preventDefault();
+
+    if (selectedTags.includes(this.innerText)) {
+      selectedTags = selectedTags.filter((t) => t !== this.innerText);
+    } else {
+      selectedTags.push(this.innerText);
+    }
+
+    this.classList.toggle('selected');
+    hasSelectedTags = selectedTags.length > 0;
+    dispatch('change', selectedTags);
+  }
+
+  function deselectAll(event: MouseEvent) {
+    event.preventDefault();
+    selectedTags = [];
+    hasSelectedTags = false;
+    dispatch('change', selectedTags);
+  }
+
+  function createTagURL(tag: string) {
+    if (urlSearchParamKey === '') {
+      return '';
+    }
+    const nextTags = selectedTags.includes(tag)
+      ? selectedTags.filter((t) => t !== tag)
+      : selectedTags.concat([tag]);
+    const params = new URLSearchParams([[urlSearchParamKey, nextTags.join(',')]]);
+    return '?' + params.toString();
+  }
+</script>
+
+<div class="tag-box">
+  <div class="tag-title" class:hidden={hasSelectedTags}>
+    <slot name="title" />
+  </div>
+  <a href="?" class="tag-clear-button" class:hidden={!hasSelectedTags} on:click={deselectAll}>
+    <slot name="resetButton" />
```

Comment:
For slot names, I do not believe we currently have decided on a naming convention. I personally think `name="reset-button"` with a `-` (hyphen; kebab-case) looks looks nicer in HTML markup. 
```suggestion
    <slot name="reset-button" />
```

Reply:
There are currently [no occurrences](https://cs.github.com/EthanThatOneKid/acmcsuf.com?q=%3Cslot) of slots with multi-part names in this repository (this would be the first). So to be clear, we will use kebab-case for the slot names when we encounter a multi-part slot name.

Reply:
Absolutely agree. It looks nicer.